app/backend/src/couchers/servicers/api.py

```diff
@@ -412,9 +412,27 @@ def UpdateProfile(self, request, context):
             return empty_pb2.Empty()
 
     def ListFriends(self, request, context):
+        users1 = aliased(User)
+        users2 = aliased(User)
+
         with session_scope() as session:
             rels = (
                 session.query(FriendRelationship)
+                .join(
+                    users1,
+                    and_(
+                        users1.id == FriendRelationship.from_user_id,
+                        FriendRelationship.from_user_id is not context.user_id,
```

Comment:
Why is this needed? If `context.user_id` is not visible, they shouldn't make it this far, so no need to worry for that.

Also, I don't think `is not` works here, should be `!=` so SA can overload it.

[Unrelated note: I believe it's often a bad idea to have expressions in the join condition for SQL performance reasons, it trashes the query planner's ability to use indexes and often causes full table scans which are very slow with more rows.]

Reply:
I thought it would be faster, but If it's slower then ya, I'll take it out.

Reply:
I don't know if it's faster, but I think it's much clearer from a code maintainability POV to take it out